src/plugins/intel_gpu/tests/unit/test_cases/border_gpu_test.cpp

```diff
@@ -1825,5 +1825,90 @@ TEST_P(border_dynamic_test, border_dynamic_test) {}
 INSTANTIATE_TEST_SUITE_P(border_dynamic_test,
                          border_dynamic_test,
                          ::testing::ValuesIn(dynamic_params));
-};  // namespace
 
+TEST(border_gpu, basic_zero_input_dynamic) {
+    auto& engine = get_test_engine();
+
+    ov::PartialShape zero_input_static = {0, 1};
+    auto input = engine.allocate_memory({zero_input_static, data_types::f32, format::bfyx});
```

Comment:
As I remember that may cause crash on some platforms due to attempt to allocate 0 bytes for USM memory. So you'd better allocate at least 1 byte and reinterpret it to required layout with zero dim.